cmake: add support for transitive ZLIB target

skirk opened the pull request:
Create transitive CMake targets for ZLIB similarly to OpenSSL. The changes follow the convention as in #3055.

skirk commented:
This still also needs accompanying changes to CMake/curl-config.cmake.in. I will add that to the PR later today.

skirk commented:
I would like to fix the appveyor build, but I fail to understand what's the issue from the logs. Any ideas?

bagder commented:
I think the appveyor failure is just a flaky test and not your fault. I'll retrigger the build.

snikulov commented:
@skirk Could you please elaborate more what issue you've tried to solve?
Are any particular troubles without those changes?

snikulov commented on CMake/curl-config.cmake.in:
```diff
@@ -1,9 +1,15 @@
 
-if("@USE_OPENSSL@")
+if("@USE_OPENSSL@" OR "@CURL_ZLIB@" )
     include(CMakeFindDependencyMacro)
```
I believe somebody will add another dependency shortly.
So why not include CMakeFindDependencyMacro unconditionally? 

```
include(CMakeFindDependencyMacro)
if("@USE_OPENSSL@")
  find_dependency(OpenSSL "@OPENSSL_VERSION_MAJOR@")
endif()
if("@USE_ZLIB@")
  find_dependency(ZLIB "@ZLIB_VERSION_MAJOR@")
endif()

```

snikulov commented on CMake/curl-config.cmake.in:
```diff
@@ -1,9 +1,15 @@
-    find_dependency(OpenSSL "@OPENSSL_VERSION_MAJOR@")
+    if ("@USE_OPENSSL@")
+        find_dependency(OpenSSL "@OPENSSL_VERSION_MAJOR@")
+    endif()
```
AFAIR, indentation used in curl project is 2 spaces. Could you please fix? Looks like it has lurked from the previous review.

skirk commented:
The commit above implements the requested changes. I also noticed the HAVE_ZLIB variable wasn't used anywhere so I changed it to USE_ZLIB to be consistent with the USE_OPENSSL. 

Also removed the unnecessary quotes from the @variables so the results after configure_file is:
 if(ON) 
rather than 
if("ON")

bagder commented:
Thanks!
